app/scripts/metamask-controller.js

```diff
@@ -5853,6 +5853,19 @@ export default class MetamaskController extends EventEmitter {
           this.controllerMessenger,
           'SnapController:getAll',
         ),
+        getCurrencyRate: (currency) => {
+          const rate = this.multichainRatesController.state.rates[currency];
+          const { fiatCurrency } = this.multichainRatesController.state;
+
+          if (!rate) {
+            return undefined;
+          }
+
+          return {
+            currency: fiatCurrency,
+            ...rate,
```

Comment:
```suggestion
            ...rate,
            currency: fiatCurrency,
```

Wondering if we should not let this be overwritten?